src/components/pages/devices/flyouts/deviceJobs/deviceJobMethods.js

```diff
@@ -1,20 +1,222 @@
+  render() {
+    const {
+      t,
+      onClose,
+      devices
+    } = this.props;
+    const {
+      isPending,
+      error,
+      successCount,
+      changesApplied,
+      commonMethods = []
+    } = this.state;
+
+    const summaryCount = changesApplied ? successCount : devices.length;
+    const completedSuccessfully = changesApplied && successCount === devices.length;
+    const summaryMessage = this.getSummaryMessage();
+    const methodOptions = (commonMethods || []).map(name => ({ value: name, label: name }));
+
+    return (
+      <form onSubmit={this.apply}>
+        <FormSection className="device-job-Methods-container">
+          <SectionHeader>{t('devices.flyouts.jobs.methods.title')}</SectionHeader>
+          <SectionDesc>{t('devices.flyouts.jobs.methods.description')}</SectionDesc>
+
+          <FormGroup>
+            <FormLabel>{t('devices.flyouts.jobs.methods.methodName')}</FormLabel>
+            <FormControl className="long" type="select" link={this.methodNameLink} options={methodOptions} placeholder={t('devices.flyouts.jobs.methods.methodNameHint')} clearable={false} searchable={true} errorState={!!error} />
+          </FormGroup>
+
+          <FormGroup>
+            <FormLabel>{t('devices.flyouts.jobs.jobName')}</FormLabel>
+            <div className="help-message">{t('devices.flyouts.jobs.jobNameHelpMessage')}</div>
+            <FormControl className="long" link={this.jobNameLink} type="text" placeholder={t('devices.flyouts.jobs.jobNameHint')} />
+          </FormGroup>
+
+          {
+            this.isFirmwareUpdate() &&
+            [
+              <FormGroup key="firmware-version">
+                <FormLabel>{t('devices.flyouts.jobs.methods.firmwareVersion')}</FormLabel>
+                <FormControl className="long" link={this.firmwareVersionLink} type="text" placeholder={t('devices.flyouts.jobs.methods.firmwareVersionHint')} />
+              </FormGroup>,
+              <FormGroup key="firmware-uir">
+                <FormLabel>{t('devices.flyouts.jobs.methods.firmwareUri')}</FormLabel>
+                <FormControl className="long" link={this.firmwareUriLink} type="text" placeholder={t('devices.flyouts.jobs.methods.firmwareUriHint')} />
+              </FormGroup>
+            ]
+          }
+
+          <SummarySection title={t('devices.flyouts.jobs.summaryHeader')}>
+            <SummaryCount>{summaryCount}</SummaryCount>
+            <SectionDesc>{summaryMessage}</SectionDesc>
+            {this.state.isPending && <Indicator />}
+            {completedSuccessfully && <Svg className="summary-icon" path={svgs.apply} />}
+          </SummarySection>
+
+          {
+            error &&
+            <AjaxError className="device-jobs-error" t={t} error={error} />
+          }
+          {
+            !changesApplied &&
+            <BtnToolbar>
+              <Btn svg={svgs.reconfigure} primary={true} disabled={!this.formIsValid() || isPending} type="submit">{t('devices.flyouts.jobs.apply')}</Btn>
```

Comment:
nit: Are we doing double quotes or single `'submit'`

Reply:
I'm pretty sure this one is double quotes. But yes, I find it confusing as well. Here's how I am interpreting what Stephen told me... use single quotes, except in translations.json and in html/jsx props.

Reply:
Yep we need linting :)

Reply:
1) You can only use double quotes in the translation file since it is just json. If you try to use single quotes translations will all fail.
2) In general, use single quotes in all JavaScript and JavaScript expressions. The exception is if the string itself has characters that make double quotes useful. e.g. `"if a string as single quote like it's and you want to avoid escaping it"`
3) Use double quotes inside all JSX. A major motivation for this is JSX is used in place of HTML and HTML uses double quotes for static attributes.

Example, 
`<Component title="This is a JSX string" />`
`<Component title={condition ? 'This is a javascript expression' : 'inside JSX' } />`